### ISSUE TITLE
zebra striping in all iv combinations fail if item goes off screen and back

### ISSUE THREAD

**Issue by nahojjjen** [farkam135/GoIV#305]
![image](https://cloud.githubusercontent.com/assets/8545451/18094823/11da8a12-6ed5-11e6-9090-ae1a8867e392.png)

**Comment by sarav** [comment]
Feature, not a bug /s

P.S: I think we should make it zebra striped by default? Looks nicer to read.

**Comment by nahojjjen** [comment]
I merged it in, even if it's a little buggy, because even with this bug it's still superior to the old implementation.

**Comment by oskomonzon** [comment]
@sarav's got my back! I owe you, man.

Seriously, tho. No clue what could be the issue there, the code is pretty straightforward. I'll give it a look once I'm back home, but I'm open to suggestions.

**Comment by sarav** [comment]
Why did you close it? Not reproducible on your device too?

Osko, not sure how I got your back or why you own me anything, but I'll take it ;)

**Comment by Blaisorblade** [comment]
@sarav He pushed a fix ;-)

**Comment by sarav** [comment]
Duh! Thanks. Checking at work during a break and didn't notice the commit.

**Comment by oskomonzon** [comment]
@sarav it was caused by the way the RecyclerView recycles its views (Made the same pun to @nahojjjen). If you don't provide the two sides of a conditional, some times it messes up. I was not able to reproduce the bug on my device (as I was at work and only had one phone to run the test). But through the power of teamwork we ran the test on @nahojjjen's end.

I'm leaving the note here in case someone runs into a similar issue.
TLDR: When working with RecyclerViews is recommended to specify both sides of a conditional sentece.

**Comment by Blaisorblade** [comment]
> TLDR: When working with RecyclerViews is recommended to specify both sides of a conditional sentence.

If I may rephrase: since RecyclerView recycles views, when updating them don't assume they're fresh but reset anything you changed.
